sim/midas/src/main/cc/emul/mm.cc

```diff
@@ -74,7 +77,7 @@ void mm_magic_t::tick(bool reset,
 
                       bool w_valid,
                       uint64_t w_strb,
-                      void *w_data,
+                      const std::vector<uint32_t> &w_data,
```

Comment:
I'm a big fan of typing things that are currently `void` but is there a reason to type this `uint32_t` instead of `uint8_t`?  

I'd prefer to also get rid of the C-style cast in the usages of `w_data.data()`.

Reply:
This is roughly what the original code was reading out. Will change in a subsequent PR.